install-server.sh

```diff
@@ -64,6 +65,13 @@ cmdline() {
         read -rp "Please provide the path of the runner private key: " PRIVATE_KEY </dev/tty
     fi
 
+    # Replace tilde with the current home:
+    private_key_absolute_path=$(realpath -q -e "$PRIVATE_KEY" || {
+         echo "Path '${PRIVATE_KEY}' to private SSH key does not exist, please try again."
+         exit 1
+    })
+    PRIVATE_KEY="$private_key_absolute_path"
```

Comment:
```suggestion
    PRIVATE_KEY="${PRIVATE_KEY/#\~/$HOME}"
    private_key_absolute_path=$(realpath -q -e "${PRIVATE_KEY}") || {
        echo "Path '${PRIVATE_KEY}' to private SSH key does not exist, please try again."
        exit 1
    }
    PRIVATE_KEY="$private_key_absolute_path"
```

@tomschr almost there! I needed to combine your solution as well since the tilde wasn't being expanded with `realpath`.
Tilde expansion doesn't work if the string starts with quotes. The output of `read` is wrapped around quotes so this is the easiest way. You can test it by doing `realpath -q -e "~/.ssh/id_rsa"` vs `realpath -q -e ~/.ssh/id_rsa`. See also here: https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_01

I tested it and this should cover all the cases.

Reply:
Thanks @fabriziosestito! I've added now your code.
Interesting findings. I wasn't aware of the issue with the quote. Great you found out! :+1: